WooCommerce/Classes/ViewRelated/Products/Add Product/ProductFactory.swift

```diff
@@ -0,0 +1,85 @@
+import Yosemite
+
+/// Creates a new product given a set of parameters.
+///
+struct ProductFactory {
+    /// Creates a new product (does not exist remotely) for a site ID of a given type.
+    ///
+    /// - Parameters:
+    ///   - type: The type of the product.
+    ///   - siteID: The site ID where the product is added to.
+    func createNewProduct(type: ProductType, siteID: Int64) -> Product {
+        switch type {
+        case .simple, .grouped, .variable, .affiliate:
+            return createEmptyProduct(type: type, siteID: siteID)
+        default:
+            fatalError("Unable to create product of type: \(type)")
```

Comment:
I was checking that the `ProductType` enum has a `.custom` case for extensions. With that in mind, wouldn't it be better for this function to throw an error or return nil instead of crashing the app?

Reply:
`.custom` is a possible case for editing and not creating, but maybe some day this won't be true. Updated to return nil in: https://github.com/woocommerce/woocommerce-ios/pull/2802/commits/5dd038da898e1407f480e277ede5d555eababe9e